.github/workflows/tests.yml

```
@@ -24,91 +30,134 @@ jobs:
+          - os: windows-2019
+            architecture: x86
+            python: 3.6
+            geos: 3.6.5
+            numpy: 1.13.3
+          - os: windows-2019
+            architecture: x86
+            python: 3.9
+            geos: 3.10.1
+            numpy: 1.19.5
 
     env:
       GEOS_VERSION: ${{ matrix.geos }}
+      GEOS_VERSION_SPEC: ${{ matrix.geos }}
       GEOS_INSTALL: ${{ github.workspace }}/geosinstall/geos-${{ matrix.geos }}
+      GEOS_BUILD: ${{ github.workspace }}/geosbuild
 
     steps:
-      - uses: actions/checkout@v2
+      - name: Correct slashes in GEOS_INSTALL (Windows)
+        run: |
+          echo 'GEOS_INSTALL=${{ github.workspace }}\geosinstall\geos-${{ matrix.geos }}' >> $GITHUB_ENV
+          echo 'GEOS_BUILD=${{ github.workspace }}\geosbuild' >> $GITHUB_ENV
+        if: ${{ matrix.os == 'windows-2019' }}
+
+      - name: Checkout Shapely
+        uses: actions/checkout@v2
+
+      - name: Checkout GEOS (main)
+        uses: actions/checkout@v2
+        with:
+          repository: libgeos/geos
+          ref: main
+          path: ${{ env.GEOS_BUILD }}
+        if: ${{ matrix.geos == 'main' }}
+
+      - name: Put the latest commit hash in the cache token for GEOS main
+        run: |
+          echo "GEOS_VERSION_SPEC=$(git rev-parse HEAD)" >> $GITHUB_ENV
+        working-directory: ${{ env.GEOS_BUILD }}
+        if: ${{ matrix.geos == 'main' }}
 
       - name: Set up Python ${{ matrix.python }}
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python }}
+          architecture: ${{ matrix.architecture }}
 
       - name: Cache GEOS and pip packages
         uses: actions/cache@v2
         with:
-          key: ${{ runner.os }}-geos-${{ matrix.geos }}
+          key: ${{ matrix.os }}-${{ matrix.architecture }}-geos-${{ env.GEOS_VERSION_SPEC }}-${{ hashFiles('ci/install_geos.sh') }}
           path: |
             ~/.cache/pip
             ${{ github.workspace }}/geosinstall
 
-      - name: Install GEOS
-        shell: bash
-        run: |
-          ./ci/install_geos.sh
-          cd ${{ github.workspace }}
+      - name: Activate MSVC
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: ${{ matrix.architecture }}
+        if: ${{ matrix.os == 'windows-2019' }}
 
-      - name: Set environment variables for PyGEOS installation
-        shell: bash
+      - name: Install GEOS
         run: |
-          echo "GEOS_CONFIG=$GEOS_INSTALL/bin/geos-config" >> $GITHUB_ENV
+          bash ci/install_geos.sh
 
       - name: Install python dependencies
-        shell: bash
         run: |
-          pip install --disable-pip-version-check --upgrade pip
-          pip install --upgrade wheel
-          pip install git+https://github.com/pygeos/pygeos.git -v
+          python -m pip install --disable-pip-version-check --upgrade pip
+          pip install --upgrade wheel setuptools
           if [ "$GEOS_VERSION" = "main" ]; then
```

The condition `[ "$GEOS_VERSION" = "main" ]` coincidentally aligns with whether or not matrix.numpy is specified, but it might not be like that in the future.

```suggestion
          if [ -n "${{ matrix.numpy }}" ]; then
```